fullscreen/model/remove-last-manual.html

```diff
@@ -23,9 +23,14 @@
             assert_equals(document.fullscreenElement, last);
             assert_equals(event.target, last);
             last.remove();
+            // Because /last/ was removed from the top layer, only /first/
+            // remains and it becomes the new fullscreen element synchronously.
+            assert_equals(document.fullscreenElement, first);
             document.onfullscreenchange = t.step_func_done(function(event)
             {
-                assert_equals(document.fullscreenElement, first);
+                // When the asynchronous steps of "exit fullscreen" ran, only
```

Comment:
@upsuper, I wrote this tests because this test began failing when working on the top layer rewrite in Chromium, and it turns out that Firefox already behaves this way.

Do the comments here match the reason it behaves like this in Firefox? I think it's quite surprising, but to fix it would require one of these:
* Making "exit fullscreen" keep track of which element it is exiting for, and doing nothing if it's no longer the fullscreen element in the async part.
* Change the removing steps so that they don't invoke "exit fullscreen" at all, but instead just resize if necessary and enqueue an event.
* Others?

WDYT?